### ISSUE TITLE
Write React Components

### ISSUE THREAD

**Issue by ovflowd** [nodejs/api-docs-tooling#10]
This is the issue that requires `nodejs/nodejs.org` to be already as a monorepo or have a way that we can import the React Components created from there;

We would also need to add the API Docs-specific Components to this repository within the `mdx-react` generator folder. Layouts (header, footer, navigation) must probably exist within the `react-web` generator (?).. I assume that a good chunk can be reused from the Node.js Website ones.

The Figma design is available here: https://www.figma.com/design/pu1vZPqNIM7BePd6W8APA5/Node.js?node-id=422-7923

**Comment by Avinashshiyani** [comment]
I can help you with that

**Comment by AugustinMauroy** [comment]
Update on that:

- [x] `nodejs/nodejs.org` as a monorepo
- [ ] i18n package 
- [ ] UI package

Once we've created the UI package, we can open an output suite to create the components.

**Comment by avivkeller** [comment]
#273 currently requires the following components[^1]:

* `<CircularIcon color={...} symbol={...} />`

![Image](https://github.com/user-attachments/assets/f3591b31-6539-4bc3-92da-43b7c26997ea)

* `<ChangeHistory changes={...} />`, where `changes` is an array of objects in the form:
  `{ versions: ["vX.Y.Z", "vX.Y.Z"], url: "https://url/to/pr", label: "description of change" }`

![Image](https://github.com/user-attachments/assets/d5ae9931-0155-47b5-aba8-734d500e554a)

[^1]: The names, and props they accept, can be changed as needed

**Comment by avivkeller** [comment]
> For `CodeBox` and `CodeTabs` make them living in `@node-core/ui-component`

No, we have base implementations there. The ones here should simply be a wrapper around those matching the wrappers we use on the website. (For example, adding the copy/paste functionality)

**Comment by ovflowd** [comment]
Good stuff! Thanks for documenting this!

**Comment by avivkeller** [comment]
All react components have been written (one still needs to land in ui components). this'll close once they are all merged into this repository in the linked PR.

I've self assigned to discourage users from taking on this issue, since it's all but completed.
